changelog: release notes for v0.2.0 (#474)
celestiaorg/celestia-node: CHANGELOG.md

# CHANGELOG
## v0.2.0 | 2022-02-23

> The second minor release focuses on the stability and robustness of existing features via minor refactorings, mainly for

Reviewer: > via minor refactorings 

I think we can omit `minor` as down below we are referring to a "almost rewritten from scratch with a better design"

Author: Can you make a suggestion?

Reviewer: ```suggestion
The second minor release focuses on the stability and robustness of existing features via refactorings, mainly for
```

headers synchronization and storing, while fixing the DA network segregation by providing bootstrap peers, allowing you to
run a Light Node with zero effort. The release also includes a few breaking changes.

### Highlights
#### Hardcoded bootstrap peers

> Bootstrappers are essential backbone peers that every other peer is connected to be part of the DA p2p network.

Reviewer 2: they don't necessarily *need* to be connected to a bootstrapper, it's just a default.

this might be confusing

Author: They have to be connected, otherwise they are disconnected from our network. What you wanted to say is probably that they don't have to use them as trusted peers to sync from - yes, but the connection to them should still present.

Author: We configure that auto connectivity here: https://github.com/celestiaorg/celestia-node/blob/main/node/p2p/routing.go#L43. This line ensures a DHT is connected to the network through the bootstrappers.

Reviewer 2: Yeah I guess the language came off a bit confusing like you had to be connected to the default bootstrap peers to be connected to the network. Lemme make a suggestion

Additionally, they serve chain headers and shares for everyone, so any other node joining the network can use them to
synchronize headers and perform Data Availability Sampling. Now it's unnecessary to maintain your own Bridge(previously
Full) Node to run a Light Node, as now it will rely on bootstrappers by default.

#### Zero effort Light Node
Now becoming a valuable network validator is only two commands away. Simply run `celestia light init` followed by `celestia
light start`, and you are good to go.

#### Test Swamp

> The release comes with a new internal testing infrastructure for Celestia Node called test Swamp, aimed to provide an

Reviewer 2: ```suggestion
The release comes with a new internal testing infrastructure for Celestia Node called Test Swamp, aimed to provide an
```

ergonomic framework for integrational testing of `celestia-node`. It allows simulating a network on which we can test various
high-level scenarios.

#### Header Synchronization
It was almost rewritten from scratch with a better design which:

> * Tolerates network disconnections and primary network interface changes(especially useful for Light Node running on

Reviewer: ```suggestion
* Tolerates network disconnections
* Tolerates primary network interface changes
  * both above are useful for Light Node running on laptops/mobiles
```

  laptops/mobiles)
* Optimizes bandwidth and IO usage, subsequently speeding up the synchronization time for ~20%
* Eliminates long-standing issue with header duplicates flooding the network
* Fixes the most common issue community has faced - constant logging of __invalid headers__ error.

#### Renamed Full Node
Full Node is now called Bridge Node to emphasize its purpose of bridging the core consensus and the halo DA networks, both

> powering the Celestia project. Spoiler: Next release will come with a reincarnation of the Full Node type, which operates only
> over the DA `celestia-node` network.

Reviewer: Deserves a quote-typed formater here:
```suggestion
powering the Celestia project. 

> Spoiler: Next release will come with a reincarnation of the Full Node type, which operates only over the DA `celestia-node` network
```
> Spoiler

Author: True.

# CHANGELOG
## v0.2.0 | 2022-02-23
The second minor release focuses on the stability and robustness of existing features via refactorings, mainly for
headers synchronization and storing, while fixing the DA network segregation by providing bootstrap peers, allowing you to
run a Light Node with zero effort. The release also includes a few breaking changes.

### Highlights
#### Hardcoded bootstrap peers

> Bootstrappers are essential backbone peers that every other peer is connected to be part of the DA p2p network.

Reviewer 2: ```suggestion
Bootstrappers are necessary for peers to connect to when they come online in order to be part of the DA p2p network.
```

# CHANGELOG
## v0.2.0 | 2022-02-23
The second minor release focuses on the stability and robustness of existing features via refactorings, mainly for
headers synchronization and storing, while fixing the DA network segregation by providing bootstrap peers, allowing you to
run a Light Node with zero effort. The release also includes a few breaking changes.

### Highlights
#### Hardcoded bootstrap peers
Bootstrappers are essential backbone peers that every other peer is connected to be part of the DA p2p network.

> Additionally, they serve chain headers and shares for everyone, so any other node joining the network can use them to

Reviewer 3: ```suggestion
Additionally, they serve block headers and shares for everyone, so any other node joining the network can use them to
```

synchronize headers and perform Data Availability Sampling. Now it's unnecessary to maintain your own Bridge(previously
Full) Node to run a Light Node, as now it will rely on bootstrappers by default.

#### Zero effort Light Node
Now becoming a valuable network validator is only two commands away. Simply run `celestia light init` followed by `celestia
light start`, and you are good to go.

#### Test Swamp
The release comes with a new internal testing infrastructure for Celestia Node called Test Swamp, aimed to provide an

> ergonomic framework for integrational testing of `celestia-node`. It allows simulating a network on which we can test various

Author: @renaynay, can you confirm the word order of the suggested change? It just doesn't feel right to me, but I am not a native English speaker and can be wrong.

> ergonomic framework for integrational testing of `celestia-node`. It allows simulating a network on which we can test various

Reviewer 3: I think the sentence is too long and could be split into two. Or it could also be shortened.

> ergonomic framework for integrational testing of `celestia-node`. It allows simulating a network on which we can test various

Reviewer 2: It's fine to have two 'fors'

high-level scenarios.

#### Header Synchronization
It was almost rewritten from scratch with a better design which:
* Tolerates network disconnections
* Tolerates primary network interface changes

>   * both above are useful for Light Node running on laptops/mobiles
>   laptops/mobiles)

Reviewer 3: ```suggestion
  * both above are useful for Light Node running on laptops or mobile phones
```

* Optimizes bandwidth and IO usage, subsequently speeding up the synchronization time for ~20%
* Eliminates long-standing issue with header duplicates flooding the network
* Fixes the most common issue community has faced - constant logging of __invalid headers__ error.

#### Renamed Full Node
Full Node is now called Bridge Node to emphasize its purpose of bridging the core consensus and the halo DA networks, both
powering the Celestia project. 

> Spoiler: Next release will come with a reincarnation of the Full Node type, which operates only over the DA `celestia-node` network

Full node operators after the update should now migrate to Bridge Node. This migration is trivial:
* Rename `.celestia-full` to `.celestia-bridge`
* Change scripts from `celestia full` to `celestia bridge`

#### Trusted Peers

> * `trusted-peer` is now `trusted-peers` flag allowing passing multiple trusted peers.

Reviewer 3: ```suggestion
* `trusted-peer` is now `trusted-peers` as the flag now allows passing multiple trusted peers.
```
